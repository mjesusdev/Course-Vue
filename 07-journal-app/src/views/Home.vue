<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
    <h1>Hello World!</h1>

    <button @click="goToDaybook" class="btn btn-primary">Primary</button>
    <button @click="goToDaybook" class="btn btn-secondary">Secondary</button>
    <button @click="goToDaybook" class="btn btn-success">Success</button>
  </div>
</template>

<script>
  // @ is an alias to /src
  export default {
    methods: {
      goToDaybook() {
        this.$router.push({ name: 'no-entry'});
      }
    }
  }
</script>
