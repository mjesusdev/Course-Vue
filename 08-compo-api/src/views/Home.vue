<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">

    <h1>Reactive vs Ref</h1>
    <h3>{{ ironman }}</h3>
    <h3>{{ hulk }}</h3>

    <button @click="changeIronman">Change Ironman</button>
    <button @click="changeHulk">Change Hulk</button>

    <h3>Ironman: {{ ironmanIsAlive }}</h3>
    <h3>Hulk: {{ hulkIsAlive }}</h3>
    <button @click="changeStatus">Change status</button>
  </div>
</template>

<script>
  import { ref, reactive } from 'vue'

  export default {
      name: 'Home',
      components: {

      },
      setup() {
          const ironman = ref({ name: 'Tony', age: 50 })
          const hulk    = reactive({ name: 'Bruce', age: 50 })

          const ironmanIsAlive = reactive(false)
          const hulkIsAlive    = ref(true)
          
          const changeStatus = () => {
              hulkIsAlive.value = !hulkIsAlive.value
          }

          return {
            ironman,
            hulk,

            ironmanIsAlive,
            hulkIsAlive,

            changeStatus,
            changeIronman: () => {
              ironman.value.name = 'Tony Stark'
              ironman.value.age  = 55 
            },
            changeHulk: () => {
              hulk.name = 'Bruce Banner'
              hulk.age  = 45
            }
          }
      }
    }
</script>
