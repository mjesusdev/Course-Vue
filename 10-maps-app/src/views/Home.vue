<template>
  <div>
    <MapView />

    <SearchBar />
    <MyLocationBtn />
    <TripDetails />
  </div>
</template>

<script lang="ts">
  import { defineComponent } from 'vue';
  import { usePlacesStore } from '../composables/usePlacesStore';

  import MapView from '../components/mapview/MapView.vue';
  import MyLocationBtn from '../components/my-location-btn/MyLocationBtn.vue';
  import SearchBar from '../components/searchbar/SearchBar.vue';
  import TripDetails from '../components/trip-details/TripDetails.vue';

  export default defineComponent({
    name: 'Home',
    components: { MapView, MyLocationBtn, SearchBar, TripDetails },
    setup() {
      usePlacesStore();
    }
  })
</script>