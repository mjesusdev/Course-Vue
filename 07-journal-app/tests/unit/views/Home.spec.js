import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home'

describe('Test in About View', () => {
    test('should run the component correctly', () => {
        const wrapper = shallowMount(Home);
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('should click in a button and redirect to "no-entry" ', () => {
        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount(Home, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')
        expect( mockRouter.push ).toHaveBeenCalled()
        expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'no-entry' })
    })
})