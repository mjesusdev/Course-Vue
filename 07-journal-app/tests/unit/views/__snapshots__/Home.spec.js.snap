// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in About View should run the component correctly 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hello World!</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button>
</div>
`;
